### ISSUE TITLE
[BUG] wrong CommandType for slash commands

### ISSUE THREAD

**Issue by Mias15** [interactions-py/interactions.py#1401]
### Library Version

5.3.0

### Describe the Bug

by logging an executed slash command via the event `on_command_completion()` each command (slash, app message, app right-click) has the CommandType value 2 (which stands for user app cmd on right-click) 

### Steps to Reproduce

Use the code below and take a look at your console

### Expected Results

that it has the proper value

### Minimal Reproducible Code

```python
@listen()
async def on_command_completion(event: events.CommandCompletion):  # Slash & Apps
    print(event.ctx._context_type)

@slash_command(name='test', sub_cmd_name='test', sub_cmd_description='testing')
async def test_cmd(ctx):
    return await ctx.send('test', ephemeral=True)

@message_context_menu(name='message app')
async def estt_mcm(ctx: ContextMenuContext):
    return await ctx.send('test', ephemeral=True)

@user_context_menu(name='user app')
async def test_ucm(ctx: ContextMenuContext):
    return await ctx.send('test', ephemeral=True)
```

### Checklist

- [X] I have searched the open issues for duplicates.
- [X] I have shown the entire traceback, if possible.
- [X] I have removed my token from display, if visible.
- [X] I have attempted to debug this myself, and I believe this issue is with the library

**Comment by LordOfPolls** [comment]
This isn't a library bug, this is a discord bug. The library does no processing on the payload from discord - that value is exactly what discord is sending
![image](https://github.com/interactions-py/interactions.py/assets/22540825/67bcc5dc-0d7c-4416-b457-3a13575d8276)

**Comment by LordOfPolls** [comment]
Thought I'd come back, as my previous reply was fairly blunt and unhelpful (hence deleted) 

The value you are seeing relates to this 
```python

class InteractionType(CursedIntEnum):
    """The type of interaction received by discord."""

    PING = 1
    APPLICATION_COMMAND = 2
    MESSAGE_COMPONENT = 3
    AUTOCOMPLETE = 4
    MODAL_RESPONSE = 5
 ```
 
 Not what you're assuming:
 ```python
 class CommandType(CursedIntEnum):
    """The interaction commands supported by discord."""

    CHAT_INPUT = 1
    """Slash commands; a text-based command that shows up when a user types `/`"""
    USER = 2
    """A UI-based command that shows up when you right click or tap on a user"""
    MESSAGE = 3
    """A UI-based command that shows up when you right click or tap on a message"""
 ```
 
 The latter isn't really used in user-exposed settings, and the lib only uses CommandType to determine what kind of ContextMenu to dispatch. From memory, I believe CommandType is only exposed in `ContextMenu` context objects. An argument could be made to expose it more, but arguably `isisnstance` is far more convinient 
 
 tldr: By this, type `2` is correct, as the type is `InteractionType.APPLICATION_COMMAND `

**Comment by Mias15** [comment]
Alright, thanks for the clarification. But then how do you figure out which type was used?

**Comment by silasary** [comment]
Just use isinstance() on the context.
